improver/categorical/decision_tree.py

```diff
@@ -639,28 +641,36 @@ def compare_array_to_threshold(
         Args:
             arr
             comparator:
-                One of  '<', '>', '<=', '>='.
+                One of  '<', '>', '<=', '>=','is_masked'.
             threshold
 
         Returns:
             Array of booleans.
 
         Raises:
-            ValueError: If comparator is not one of '<', '>', '<=', '>='.
+            ValueError: If comparator is not one of '<', '>', '<=', '>=','is_masked'.
         """
         if comparator == "<":
-            return arr < threshold
+            result = arr < threshold
         elif comparator == ">":
-            return arr > threshold
+            result = arr > threshold
         elif comparator == "<=":
-            return arr <= threshold
+            result = arr <= threshold
         elif comparator == ">=":
-            return arr >= threshold
+            result = arr >= threshold
+        elif comparator == "is_masked":
+            if np.ma.is_masked(arr):
+                result = arr.mask
+            else:
+                result = arr != arr
         else:
             raise ValueError(
                 f"Invalid comparator: {comparator}. "
-                "Comparator must be one of '<', '>', '<=', '>='."
+                "Comparator must be one of '<', '>', '<=', '>=','is_masked'."
             )
```

Comment:
I had a go at refactoring this to remove all the `elif`s - the end result is prettier, but not actually any easier to read, so leave it as it is.